packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,6 +1090,86 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  private onBootCleanup() {
+    this.initApprovalsOfUnapprovedTransactions();
+    this.loadGasValuesOfUnapprovedTransactions();
+    this.submitApprovedTransactions();
+  }
+
+  /**
+   * Create approvals for all unapproved transactions on current chain.
+   */
+  private initApprovalsOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      this.processApproval(transactionMeta, {
+        shouldShowRequest: false,
+      }).catch((error) => {
+        /* istanbul ignore next */
+        console.error('Error during persisted transaction approval', error);
+      });
+    }
+  }
+
+  /**
+   * Update the gas values of all unapproved transactions on current chain.
+   */
+  private async loadGasValuesOfUnapprovedTransactions() {
```

Comment:
```suggestion
  private async loadGasValuesForUnapprovedTransactions() {
```

Reply:
Fixed in https://github.com/MetaMask/core/pull/1916/commits/8b11fe7ca9352e56a3f261ba3c88fff69b0f06a9